eclair-node/src/main/scala/fr/acinq/eclair/api/directives/ExtraDirectives.scala

```diff
@@ -14,37 +14,53 @@
  * limitations under the License.
  */
 
-package fr.acinq.eclair.api
+package fr.acinq.eclair.api.directives
 
+import fr.acinq.eclair.api.serde.JsonSupport.serialization
+import akka.http.scaladsl.common.{NameReceptacle, NameUnmarshallerReceptacle}
 import akka.http.scaladsl.marshalling.ToResponseMarshaller
 import akka.http.scaladsl.model.StatusCodes.NotFound
 import akka.http.scaladsl.model.{ContentTypes, HttpResponse}
 import akka.http.scaladsl.server.{Directive1, Directives, MalformedFormFieldRejection, Route}
 import fr.acinq.bitcoin.ByteVector32
 import fr.acinq.bitcoin.Crypto.PublicKey
 import fr.acinq.eclair.ApiTypes.ChannelIdentifier
-import fr.acinq.eclair.api.FormParamExtractors._
-import fr.acinq.eclair.api.JsonSupport._
+import fr.acinq.eclair.api.serde.FormParamExtractors._
+import fr.acinq.eclair.api.serde.JsonSupport._
 import fr.acinq.eclair.payment.PaymentRequest
 import fr.acinq.eclair.{MilliSatoshi, ShortChannelId}
 
 import scala.concurrent.Future
+import scala.concurrent.duration.DurationInt
 import scala.util.{Failure, Success}
 
 trait ExtraDirectives extends Directives {
 
   // named and typed URL parameters used across several routes
-  val shortChannelIdFormParam = "shortChannelId".as[ShortChannelId](shortChannelIdUnmarshaller)
-  val shortChannelIdsFormParam = "shortChannelIds".as[List[ShortChannelId]](shortChannelIdsUnmarshaller)
-  val channelIdFormParam = "channelId".as[ByteVector32](sha256HashUnmarshaller)
-  val channelIdsFormParam = "channelIds".as[List[ByteVector32]](sha256HashesUnmarshaller)
-  val nodeIdFormParam = "nodeId".as[PublicKey]
-  val nodeIdsFormParam = "nodeIds".as[List[PublicKey]](pubkeyListUnmarshaller)
-  val paymentHashFormParam = "paymentHash".as[ByteVector32](sha256HashUnmarshaller)
-  val fromFormParam = "from".as[Long]
-  val toFormParam = "to".as[Long]
-  val amountMsatFormParam = "amountMsat".as[MilliSatoshi]
-  val invoiceFormParam = "invoice".as[PaymentRequest]
+  val shortChannelIdFormParam: NameUnmarshallerReceptacle[ShortChannelId] =
```

Comment:
nit: can you put these all back to a single line?

Reply:
Think I need to change some more Idea settings